llvm/lib/Target/AMDGPU/AMDGPUInstructions.td

```diff
@@ -169,29 +169,22 @@ def brtarget   : Operand<OtherVT>;
 class HasOneUseUnaryOp<SDPatternOperator op> : PatFrag<
   (ops node:$src0),
   (op $src0),
-  [{ return N->hasOneUse(); }]> {
-
-  let GISelPredicateCode = [{
-    return MRI.hasOneNonDBGUse(MI.getOperand(0).getReg());
-  }];
+  [{ return true; }]> {
```

Comment:
Can't we just remove the argument entirely?

Reply:
Yes, the custom predicates should go